app/models/plate.rb

```diff
@@ -37,13 +37,19 @@ class Plate < Labware
   belongs_to :plate_purpose, inverse_of: :plates
   belongs_to :purpose, foreign_key: :plate_purpose_id
   has_many :wells, inverse_of: :plate, foreign_key: :labware_id do
-    # Build empty wells for the plate.
     def construct!
-      plate = proxy_association.owner
-      plate.maps.in_row_major_order.ids.map do |location_id|
-        { map_id: location_id }
-      end.tap do |wells|
-        plate.wells.create!(wells)
+      transaction do
+        plate = proxy_association.owner
+        plate.maps.in_row_major_order.ids.map do |location_id|
+          { map_id: location_id }
+        end.tap do |wells|
+          plate.wells.import(wells)
+          ids = plate.wells.ids
+          well_type = Well.base_class.name
+          WellAttribute.import(ids.map { |well| { well_id: well } })
+          Uuid.import(ids.map { |well| { resource_id: well, resource_type: well_type, external_id: Uuid.generate_uuid } })
+          ids.each { |id| Warren::QueueBroadcastMessage.new(class_name: 'Well', id: id).queue(Warren.handler) }
+        end
```

Comment:
Do you have to add all this stuff to do with WellAttribute, Uuid and Warren because this happened automatically when `create!` was used, but doesn't with `import` ?

Reply:
Yep, import doesn't trigger rails callbacks. Which is probably a good thing, as otherwise it would still end up generating 96 insert statements. But it does mean this gets a bit more complicated.